src/number_theory/bernoulli_polynomials.lean

```diff
@@ -100,9 +100,9 @@ begin
   simp_rw [polynomial.smul_monomial, mul_comm (bernoulli _) _, smul_eq_mul, ←mul_assoc],
   conv_lhs { apply_congr, skip, conv
     { apply_congr, skip,
-      rw [choose_mul ((nat.le_sub_left_iff_add_le (mem_range_le H)).1 (mem_range_le H_1))
-        (le.intro rfl), add_comm x x_1, nat.add_sub_cancel, mul_assoc, mul_comm, ←smul_eq_mul,
-        ←polynomial.smul_monomial], },
+      rw [← nat.cast_mul, choose_mul ((nat.le_sub_left_iff_add_le $ mem_range_le H).1
+        $ mem_range_le H_1) (le.intro rfl), nat.cast_mul, add_comm x x_1, nat.add_sub_cancel,
```

Comment:
I suspect `le.intro rfl` could be `le_rfl`, and perhaps the `choose_mul` could be moved into a separate `have` for readability.

Reply:
Oh whoops! Your comment comes a bit late. I must say I didn't spot that golf.